samples/bluetooth/central_hr_coded/src/main.c

```diff
@@ -0,0 +1,279 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+
+static void start_scan(void);
+
+static struct bt_conn *default_conn;
+
+static struct bt_uuid_16 uuid = BT_UUID_INIT_16(0);
+static struct bt_gatt_discover_params discover_params;
+static struct bt_gatt_subscribe_params subscribe_params;
+
+static uint8_t notify_func(struct bt_conn *conn,
+		struct bt_gatt_subscribe_params *params,
+		const void *data, uint16_t length)
+{
+	if (!data) {
+		printk("[UNSUBSCRIBED]\n");
+		params->value_handle = 0U;
+		return BT_GATT_ITER_STOP;
+	}
+
+	printk("[NOTIFICATION] data %p length %u\n", data, length);
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static uint8_t discover_func(struct bt_conn *conn,
```

Comment:
This code contains a partial implementation of a Heart Rate Client. The NCS approach is to extract such code into separate re-usable components, e.g.  subsys/bluetooth/services/hids_c.c. Have you considered implementing a Heart Rate Client - maybe in a separate PR?

Reply:
Well, not yet because I'm just using it is a sample and I do not use Heart Rates for other projects. Alternatively, I could rewrite the code and make a BAS client out of it and use the existing BAS implementation.

Reply:
OK, let's leave the HR client as it is. It may be refactored in a separate PR at a later stage.